src/Chapter3.hs

```diff
@@ -971,6 +1062,23 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _ = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sun = Mon
+nextDay d = succ d
+
+daysToParty :: WeekDay -> Int
+daysToParty day =  iter day 0 
+  where
+    iter Fri x = x
+    iter d x = iter (nextDay d) (x + 1)
```

Comment:
Great implementation! Alternatively, you can use the `fromEnum` method of the `Enum` typeclass to implement this function more efficiently :slightly_smiling_face: